### DISCUSSION TITLE
Add saturationModifier to FoodLevelChangeEvent

### DISCUSSION THREAD

**Discussion by aerulion** [PaperMC/Paper#10529] [Ideas]
### Is your feature request related to a problem?

I'm experimenting with custom food items and trying to implement new food types. When listening to the FoodLevelChangeEvent I'm able to modify the resulting food level, but there is no simple way to modify the amount of saturation the player receives.

### Describe the solution you'd like.

Adding these two methods to the FoodLevelChangeEvent:

- FoodLevelChangeEvent#getSaturationModifier()
- FoodLevelChangeEvent#setSaturationModifier(float saturationModifier)

### Describe alternatives you've considered.

Listening to the event and doing the saturation calculation on my own, then setting the resulting value after the event has been processed.

**Comment by Owen1212055** [comment]
Working on implementing this perhaps, I'm curious would something like this also work as a separate event? 

The issue is that this event is for food level changing only, and that doesn't always mean that saturation changes too. 
It may be optimal for there to be a new event, like ``SaturationLevelChangeEvent`` which would pretty much cover this and would allow you to change things like how saturation decays (food level change event is not fired here).

**Comment by aerulion** [comment]
I guess a new event could be useful in some cases, but that wouldn't be the ideal solution for implementing custom food in my opinion. The food level and saturation are linked together, and i would need to listen to both events to first change the food level, and then the saturation in another event. This would also mean doing the checks to the item twice. Another possible to this would be something like a PlayerEatEvent, which then would provide access to the item, the food level and the saturation.
